op-node/node/safedb/safedb.go

```diff
@@ -0,0 +1,112 @@
+package safedb
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"math"
+	"sync/atomic"
+
+	"github.com/cockroachdb/pebble"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+var (
+	ErrNotFound     = errors.New("not found")
+	ErrInvalidEntry = errors.New("invalid db entry")
+)
+
+type SafeDB struct {
+	log log.Logger
+	db  *pebble.DB
+
+	writeOpts *pebble.WriteOptions
+
+	closed atomic.Bool
+}
+
+func KeyL1BlockNum(num uint64) []byte {
+	key := make([]byte, 0, 9)
+	key = append(key, 0)
```

Comment:
Making the key-prefix-byte a constant, and adding a comment about the namespace expanding, may be useful for future us. And any iterator should use the key-prefix byte constant, to select the right range of keys.